packages/eslint-mdx/src/traverse.ts

```diff
@@ -7,7 +9,7 @@ import {
   isOpenTag,
   isSelfClosingTag,
 } from './regexp'
-import type { Node, Parent, TraverseOptions, Traverser } from './types'
+import type { Jsx, Node, Parent, TraverseOptions, Traverser } from './types'
 
 export class Traverse {
```

Comment:
there may also be an opportunity to refactor this to `unist-util-visit` or `unist-util-select`
I've been cautious to do so because I still don't fully understand what `combineJsxNodes` does.
It combines nodes of course :wink: 
But there also seem to be some normalization steps, conditions around when it is done (based off an `offset`), that I haven't fully pieced together what's happening. :thinking:

Reply:
For example some of the filtering could be done with:

```ts
const isParent = convert((node: Node): node is Parent => "children" in node);

visit(
  node,
  (node: Node): node is Parent =>
    isParent &&
    !is(node, "root") &&
    node.children.filter(({ type }) => type === "jsx").length > 1,
  (node: Parent) => {
    // node is a Parent, not a Root node, and has multiple child Jsx nodes
  }
);
```

Reply:
Good to refactor, maybe in another PR.

Reply:
> But there also seem to be some normalization steps, conditions around when it is done (based off an offset), that I haven't fully pieced together what's happening. 🤔

You can check the linked issues in the comments, if it's still confusing I can explain and can you help to comment from another point of view?